core/src/main/java/org/sonarsource/sonarlint/core/http/HttpClientManager.java

```diff
@@ -63,48 +60,55 @@ public class HttpClientManager {
   private static final Timeout RESPONSE_TIMEOUT = Timeout.ofMinutes(10);
   private final SonarLintLogger logger = SonarLintLogger.get();
   private final SonarLintClient client;
-  private CloseableHttpAsyncClient sharedClient;
+  private final CloseableHttpAsyncClient sharedClient;
 
-  public HttpClientManager(SonarLintClient client, String userAgent) {
+  public HttpClientManager(SonarLintClient client, String userAgent, Path sonarlintUserHome) {
     this.client = client;
-    this.sharedClient = HttpAsyncClients.custom()
-      .setConnectionManager(
-        PoolingAsyncClientConnectionManagerBuilder.create()
-          .setTlsStrategy(ClientTlsStrategyBuilder.create()
-            .setTlsDetailsFactory(parameter -> new TlsDetails(parameter.getSession(), parameter.getApplicationProtocol())).build())
-          .build())
-      .addResponseInterceptorFirst(new RedirectInterceptor())
-      .setUserAgent(userAgent)
-      // SLI-629 - Force HTTP/1
-      .setVersionPolicy(HttpVersionPolicy.FORCE_HTTP_1)
-      // proxy settings
-      .setRoutePlanner(new SystemDefaultRoutePlanner(new ProxySelector() {
+    var sslFactoryBuilder = SSLFactory.builder()
+      .withDefaultTrustMaterial()
+      .withSystemTrustMaterial()
+
+      .withTrustMaterial(new X509TrustManager() {
```

Comment:
Minor remark since it will be likely reworked when doing the real implementation, but I think this should be moved to its own class as it will probably get more complex